api/AxiosProvider.tsx

```diff
@@ -0,0 +1,19 @@
+'use client';
+
+import React, { createContext, useContext } from 'react';
+import { AxiosInstance } from 'axios';
+import axiosInstanceDefault from './axiosInstance';
+
+const AxiosContext = createContext<AxiosInstance | null>(null);
+
+export const useAxios = () => {
+  const axiosInstance = useContext(AxiosContext);
+  if (!axiosInstance) {
+    throw new Error('useAxios must be used within an AxiosProvider');
+  }
+  return axiosInstance;
+};
+
+export const AxiosProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
```

Comment:
refactor to not use React.FC

Reply:
Also resolve conflicts

Reply:
Why not use React.FC? I see it being used all over the codebase. For example, it is used in themeContext as well.

Reply:
Just resolved the conflicts. The **E2E tests / run\_tests** failed because I have put a check for the presence of ENV variable **NEXT\_PUBLIC\_API\_BASE\_URL** before setting the endpoints. This is not an issue if project is properly setup.

Reply:
@petark7 what is not properly setup?

Reply:
@edichoska 

The check I placed in the code checks if the ENV variable '**NEXT\_PUBLIC\_API\_BASE\_URL' exists on the environment variables file, if not, it throws an error. This is what happened during the tests ran in this PR. On my end, there is no issue at all (since I have the ENV var set).**

Reply:
ahaaa e2e tests are failing.